### PR TITLE
JDK-8322235: Split up and improve LocaleProvidersRun

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which splits up the _LocaleProvidersRun_ test file for performance and maintenance reasons.

_LocaleProvidersRun_ which tests against the various Locale Providers (CLDR, HOST, SPI, etc.) was getting rather long, as all related bugs were added to the single test file. To improve maintainability, this change splits up the single test file into separate test files focused on specific areas (ex: _j.text.Format_).  The original _LocaleProvidersRun_ test file remains and is used for more general Locale Provider testing such as the adapter loading.

In addition, the previously single test file used to suffer from performance issues, as each test method had to launch a new JVM (since Locale Providers are set at Java startup time). With this change, these tests files can be ran with VM flags and not cause timeout, thus `@requires vm.flagless` is no longer needed (Tiers 6-8 tested).

Other updates
- For OS/locale specific tests, the OS/locale is now checked before (not after) launching a JVM
- Added comments for each test method

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322235](https://bugs.openjdk.org/browse/JDK-8322235): Split up and improve LocaleProvidersRun (**Sub-task** - P4)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [c754893f](https://git.openjdk.org/jdk/pull/17257/files/c754893fa40bb76bbce0b254f04338ad714ddd43)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17257/head:pull/17257` \
`$ git checkout pull/17257`

Update a local copy of the PR: \
`$ git checkout pull/17257` \
`$ git pull https://git.openjdk.org/jdk.git pull/17257/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17257`

View PR using the GUI difftool: \
`$ git pr show -t 17257`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17257.diff">https://git.openjdk.org/jdk/pull/17257.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17257#issuecomment-1876114090)

### CODE REVIEW

File: test/jdk/java/util/Locale/LocaleProvidersFormat.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 7198834 8001440 8013086 8013903 8027289 8232860
+ * @summary Test any java.text.Format Locale provider related issues
+ * @library /test/lib
+ * @build LocaleProviders
+ *        providersrc.spi.src.tznp8013086
```
Comment (naotoj) [inline test/jdk/java/util/Locale/LocaleProvidersFormat.java:30]: Although it is not needed in this test, I would list `tznp` to be built here, as the provider's meta-info includes both classes for completeness. Applies to `LocaleProvidersTimeZone` test too.

**Review by naotoj — COMMENTED** [review]
Great to see this refactoring! Much cleaner now.

File: test/jdk/java/util/Locale/LocaleProvidersFormat.java (hunk 1 above)
Comment (justin-curtis-lu) [reply test/jdk/java/util/Locale/LocaleProvidersFormat.java:31]: That makes sense, updated both the relevant test files as you suggested. Thank you for taking the time to review this.

**Review by naotoj — APPROVED** [review]

**Review by naotoj — APPROVED** [review]
Still looks good. Please modify the copyright year for `HostLocaleProviderAdapter_md.c` file before integrating

**Comment by justin-curtis-lu** [comment]
> Still looks good. Please modify the copyright year for `HostLocaleProviderAdapter_md.c` file before integrating

Totally missed that, thanks for catching. Will integrate after re-testing VM tiers.

**Pull request closed without merging**